File: 팁스타운/2017/단어_퍼즐/solve.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int solution(vector<string> strs, string t)
{
    vector<int> dp(t.size(), 987654321);
    
    for(int pos=0; pos<t.size(); pos++){
        for(auto word: strs){
            bool flag = true;
            for(int i=0; i<word.size(); i++){
                if(t[pos+i] != word[i]){
                    flag = false;
                    break;
                }
            }
            if(flag){
                if(pos == 0) dp[pos+word.size()-1] = 1;
                else dp[pos+word.size()-1] = min(dp[pos+word.size()-1], dp[pos-1]+1);
            }
        }
    }
    if(dp[t.size()-1] == 987654321) return -1;
    return dp[t.size()-1];
}

File: 깊이-너비_우선_탐색/단어_변환/solve.py
from queue import Queue

def solution(begin, target, words):
    flag = False
    
    # 아예 못만드는 경우
    for w in words:
        if w == target:
            flag = True
            break
            
    if not flag:
        return 0
    
    visited = [0 for i in range(len(words))]
    queue = Queue()
    queue.put(begin, 0)
    
    count = 0
    while not queue.empty():
        queue2 = Queue()
        while not queue.empty():
            cur = queue.get()
            
            if cur == target:
                break
                
            for i in range(len(words)):
                word = words[i]
                cnt = 0
                for j in range(len(cur)):
                    if cur[j] != word[j]:
                        cnt += 1
                if cnt == 1 and not visited[i]:
                    queue2.put(word)
                    visited[i] = True

        queue = queue2
        count += 1
        
    return count-1
